### PR TITLE
Cross-compile the kn binary for linux/s390x

### PR DESCRIPTION
**Pull request by BbolroC**
## Description

As part of the work to support the `knative` for the IBM Z system (`s390x` architecture), this PR is to make the `kn-linux-s390x` binary available in the `release`.

## Changes

* Add an additional `go build` command with `GOOS=linux GOARCH=s390x` in `hack/build.sh` and `hack/release.sh`, respectively.

## Reference

None

### CODE REVIEW

File: hack/build.sh [hunk 1]
```diff
@@ -284,6 +284,8 @@ cross_build() {
   GOOS=darwin GOARCH=amd64 go build -mod=vendor -ldflags "${ld_flags}" -o ./kn-darwin-amd64 ./cmd/... || failed=1
   echo "   🎠 kn-windows-amd64.exe"
   GOOS=windows GOARCH=amd64 go build -mod=vendor -ldflags "${ld_flags}" -o ./kn-windows-amd64.exe ./cmd/... || failed=1
+  echo "   Z  kn-linux-s390x"
```
Comment (rhuss) [inline hack/build.sh:287]:
```suggestion
  echo "   🗡️  kn-linux-s390x"
```

We love emojis ;-) What's about a dagger like the one that [Zorro](https://en.wikipedia.org/wiki/Zorro) used ?

File: hack/release.sh [hunk 2]
```diff
@@ -35,9 +35,11 @@ function build_release() {
   GOOS=darwin GOARCH=amd64 go build -mod=vendor -ldflags "${ld_flags}" -o ./kn-darwin-amd64 ./cmd/...
   echo "🚧 🎠 Building for Windows"
   GOOS=windows GOARCH=amd64 go build -mod=vendor -ldflags "${ld_flags}" -o ./kn-windows-amd64.exe ./cmd/...
+  echo "🚧 Z  Building for Linux(s390x)"
```
Comment (rhuss) [inline hack/release.sh:38]:
```suggestion
  echo "🚧 🗡️  Building for Linux(s390x)"
```

see comment above.

**Review by rhuss — COMMENTED** [review]
Looks good to me, thanks for the PR ! (a minor suggestion added).

Technically that's ok (and affair that's the second request for adding s390), so I tend to add this platform, too. I wonder whether we then should go all in and also add other popular platforms  like armv7 or arm64.

Also, we then should push multiarch OCI image for our binary.

Wdyt, @maximilien @navidshaikh and others ? Let's discuss this on the next WG call how wide we want to extend our arch support.

**Review by maximilien — APPROVED** [review]
/ok-to-test

File: hack/build.sh (hunk 1 above)
Comment (BbolroC) [reply hack/build.sh:287]: Thanks for the suggestion. I love a dagger, but let me check that with my manager first if the emoji, which does not representing the Z system, can be used in the open source. (actually I spent more time on choosing a suitable emoji than the PR itself and ended up using the alphabet, which is not a proprietary asset)

Could you please hold this PR until I get an answer for this?

Comment (rhuss) [reply hack/build.sh:287]: sure. btw, we are using also non official emoji for all the other OS, and the emoji is not meant to represent an official product logo, but just as kind of a fancy "bullet point" in the list. We also never talk about the architecture itself here, as you see it's just the name of the binary that we produce.

Comment (BbolroC) [reply hack/build.sh:287]: Yop, I have the same opinion. I am just wondering if the company thinks the same. ^^

Comment (BbolroC) [reply hack/build.sh:287]: I got the decision from the company about the emoji: just sticking to the capital z as it is. Is it okay to go with it?

**Review by maximilien — APPROVED** [review]
LGTM

**Comment by maximilien** [comment]
Looks like you need to rebase @BbolroC file: `hack/release.sh`

**Comment by BbolroC** [comment]
> Looks like you need to rebase @BbolroC file: `hack/release.sh`

Thanks for the notice. The rebase is done.

**Review by navidshaikh — APPROVED** [review]
/lgtm
/approve
could've made to v0.19 release , but we can consider this for patch release as well?
/hold for @rhuss

**Comment by rhuss** [comment]
> could've made to v0.19 release , but we can consider this for patch release as well?

+1 one for backporting this to 0.18 and 0.19

/unhold

**Pull request merged**